### PR TITLE
Make additional managed + native logging improvements

### PR DESCRIPTION
**Pull request by zacharycmontoya**
Makes planned improvements to the managed and native tracer logging:
- Both
  - Adds a new `DD_TRACE_LOG_DIRECTORY` environment variable to take precedence over `DD_TRACE_LOG_PATH`
- Native
  - Renames output file from `dotnet-profiler.log` to `dotnet-tracer-native.log`
  - Adds `Windows`/`Linux` to the native log output
  - Change the default behavior to only print out the Datadog environment variables that are set. When set to debug mode, the native logs will write all Datadog environment variables, including empty ones
- Managed
  - Adds unit tests for dynamically setting the managed tracer's log-level

One additional fix is made: Do not use the `EnvironmentHelpers` utility class in the `EnvironmentConfigurationSource` class. This creates an indirect dependency from the configuration system to the logging system.

@DataDog/apm-dotnet

### CODE REVIEW

**Comment by tonyredondo** [comment]
Now that I'm reviewing this PR I have a question, on Linux we work in the assumption that we have write access to `/var/log/datadog/dotnet/`. What happens when we don't?, A critical thing, do we crash?. Second thing, can't we use the `˜` home folder in those cases?

**Review by tonyredondo — APPROVED** [review]

File: src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp [hunk 1]
```diff
@@ -91,7 +91,10 @@ CorProfiler::Initialize(IUnknown* cor_profiler_info_unknown) {
   Info("Environment variables:");
 
   for (auto&& env_var : env_vars_to_display) {
-    Info("  ", env_var, "=", GetEnvironmentValue(env_var));
+    WSTRING env_var_value = GetEnvironmentValue(env_var);
+    if (debug_logging_enabled || !env_var_value.empty()) {
```
Comment (lucaspimentel) [inline src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp:95]: ❤️❤️❤️

**Review by lucaspimentel — APPROVED** [review]

**Comment by lucaspimentel** [comment]
> on Linux we work in the assumption that we have write access to `/var/log/datadog/dotnet/`. What happens when we don't?

We used to have a `try/catch` ([source](https://github.com/DataDog/dd-trace-dotnet/blob/96fadb461f62a92205deda75111ba869e59ba6dc/src/Datadog.Trace.ClrProfiler.Native/logging.cpp#L22-L42)). We didn't crash, but we didn't write any logs, either. I don't know what happens now since we migrated to `spdlog`. We should definitely test this.

> Second thing, can't we use the `˜` home folder in those cases?

Seems like a reasonable fallback. One concern I have is that users won't think to look there for the logs. We should make sure to update the docs and inform the Support Team about this change. Logs that can't be found are as good as no logs 😉 This will be less of an issue once we are able to include tracer logs in the Agent Flare.

**Comment by tonyredondo** [comment]
> @tonyredondo:
> 
> > on Linux we work in the assumption that we have write access to `/var/log/datadog/dotnet/`. What happens when we don't?
> 
> We used to have a `try/catch` ([source](https://github.com/DataDog/dd-trace-dotnet/blob/96fadb461f62a92205deda75111ba869e59ba6dc/src/Datadog.Trace.ClrProfiler.Native/logging.cpp#L22-L42)). We didn't crash, but we didn't write any logs, either. I don't know what happens now since we migrated to `spdlog`. We should definitely test this.
> 
> > Second thing, can't we use the `˜` home folder in those cases?
> 
> Seems like a reasonable fallback. One concern I have is that users won't think to look there for the logs. We should make sure to update the docs and inform the Support Team about this change. Logs that can't be found are as good as no logs 😉 This will be less of an issue once we are able to include tracer logs in the Agent Flare.

Ok, I've just saw the spdlog source, and it will throw an exception in the rotating sink .ctor, so I'll create a PR to handle that case.

**Comment by tonyredondo** [comment]
Done here: https://github.com/DataDog/dd-trace-dotnet/pull/1028

**Comment by zacharycmontoya** [comment]
This is the docs PR to track logging changes: https://github.com/DataDog/documentation/pull/9016

**Pull request merged**